daemon/compartment.c

```diff
@@ -676,14 +675,21 @@ compartment_sigchld_cb(UNUSED int signum, event_signal_t *sig, void *data)
 	TRACE("SIGCHLD handler called for compartment %s with PID %d",
 	      compartment_get_description(compartment), compartment->pid);
 
+	if (compartment->pid < 0) {
```

Comment:
imho we should also handle the case compartment->pid == 0 explicitly here

Reply:
We could assert that a process id is always > 0 and in compartment_new() compartment->pid is initialized with -1; and only changed as retval from clone() (-1 on error) and in cleanup cases again set to -1; Hmm I think it would be more clean way to just check  compartment->pid == -1 here.